### DISCUSSION TITLE
WiX v4.0.2 Unable to create registry values at certain paths

### DISCUSSION THREAD

**Discussion by kamillo** [wixtoolset/issues#7769] [Questions] [answered]
I'm unable to create registry values during installation at certain paths, e.g. `HKLM\SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\VisualEffects` while other registry paths works without an issue. There are no errors in the installator log file.

Adding values using command line works as well (`reg add HKLM\SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\VisualEffects /v VisualFXSetting /t REG_DWORD /d 3`)

I've observed the same behaviour for versions 4.0.0, 4.0.1 and 4.0.2

Is there any know limitations?

Minimal example that can be used to reproduce this issue (first components succesfully creates registry value but second doesn't): 
```xml
<Wix xmlns="http://wixtoolset.org/schemas/v4/wxs">
  <Fragment>
    <ComponentGroup Id="ExampleComponents" Directory="INSTALLFOLDER">
      <Component>
        <RegistryValue Root="HKLM" Key="SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\AutoplayHandlers" Name="VisualFXSetting" Type="integer" Value="3" KeyPath="yes" />
      </Component>
      <Component>
        <RegistryValue Root="HKLM" Key="SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\VisualEffects" Name="VisualFXSetting" Type="integer" Value="3" KeyPath="yes" />
      </Component>
    </ComponentGroup>
  </Fragment>
</Wix>
```

**Comment by kamillo** [comment]
The problem appears when running x86 installer on x64 system. Registry values are written but in the separate `Wow6432Node`. Unfortunately those values aren't be picked up by the system. 
In my case I wan't to install 32bit binaries/dlls to 64bit system and configure the system settings using registry values. As it seems that it is not allowed write to 64 bit registry from 32bit installer is there any better approach than creating separate installers - one (32bit) for exes/dlls and another (64bit) for registry and bundling them together?

**Reply by robmen** [reply, marked as answer]
64-bit packages can install 32-bit resources. So you don't have to create a 32-bit package unless you want to install on 32-bit OSes (which aren't common anymore but you'll know your customers better).

**Reply by kamillo** [reply]
Thanks, you are right, this looks like the right approach. There is no need for 32-bit OS support. The only downside is that every component (except 64-bit registry values) needs to be explicitly marked 32-bit (`Bitness="always32"`).
